### PR TITLE
updated wandb logger

### LINKED ISSUE #1894: Missing `close` method in `WandBLogger` and `PolyaxonLogger` ?
## 🐛 Bug description

Just found it unexpected. Not sure this is intended. There are `close` method overrides in provided loggers except for `WandBLogger` and `PolyaxonLogger`. [wandb states that wandb process have to close if we are on jupyter or colab env](https://docs.wandb.ai/integrations/huggingface#6-finish-your-w-and-b-run-notebook-only). Not sure about PolyaxonLogger.

We need to have one API to close all loggers even if users want to change using different loggers.

### PR DESCRIPTION
**Pull request by ahmedo42**
Fixes #1894

Description:

wandb logger should have close method.

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

**Review by sdesrozis — APPROVED** [review]
@ahmedo42 Thank you ! LGTM

**Comment by vfdev-5** [comment]
@ahmedo42 to close #1894 we have to add empty close to PolyaxonLogger as well and additionally it would be helpful to check Polyaxon API if they still do not require any closing method.

**Comment by ahmedo42** [comment]
> it would be helpful to check Polyaxon API if they still do not require any closing method.

I checked the docs and didn't find that they require one.

**Comment by sdesrozis** [comment]
> I checked the docs and didn't find that they require one.

@ahmedo42 Are you sure ? I think there is a `tracking.end()` to manually end a run.

**Comment by ahmedo42** [comment]
>  I think there is a `tracking.end()` to manually end a run.

Yes you're correct there is , but as far as we're using this logger it should be closed by calling  [Run.end()](https://polyaxon.com/docs/experimentation/tracking/client/#end) except when I simply call this I get the following error:

` 'Run' object has no attribute '_exit_handler' ` which entails that  the logger isn't initialized correctly perhaps?

Also I couldn't find any useful documentation for closing the logger when we use Experiment instead of Run

So wouldn't using [tracking module](https://polyaxon.com/docs/experimentation/tracking/module/) instead of Run and Experiment be more cleaner?

**Comment by sdesrozis** [comment]
Sorry I meant `Run.end()`... Btw it is necessary to dig further the issue when calling this method. Maybe trying to reproduce this issue without ignite would help to understand if it's on our side or not ?

**Comment by ahmedo42** [comment]
@sdesrozis  I filed an issue on Polyaxon and it's indeed a bug https://github.com/polyaxon/polyaxon/issues/1258#issuecomment-811511510

**Comment by sdesrozis** [comment]
@ahmedo42 Thank you! 

So we have to handle an exception inside the method `close()` as suggested by Polyaxon team. Or do a correct initialisation (another PR ?)

**Comment by ahmedo42** [comment]
@sdesrozis  I think catching the exception would be OK for now until they fix it ( even if it's a bit lazy 😄 ), if you think differently it  shouldn't be so difficult to open another PR.

**Comment by sdesrozis** [comment]
> @sdesrozis  I think catching the exception would be OK for now until they fix it ( even if it's a bit lazy 😄 )

Let's do it like that 😊

File: ignite/contrib/handlers/polyaxon_logger.py [hunk 1]
```diff
@@ -87,6 +87,8 @@ class PolyaxonLogger(BaseLogger):
                 optimizer=optimizer,
                 param_name='lr'  # optional
             )
+            # We need to close the logger when we are done
+            wandb_logger.close()
```
Comment (vfdev-5) [inline ignite/contrib/handlers/polyaxon_logger.py:91]:
```suggestion
            plx_logger.close()
```

File: ignite/contrib/handlers/polyaxon_logger.py [hunk 2]
```diff
@@ -87,6 +87,8 @@ class PolyaxonLogger(BaseLogger):
                 optimizer=optimizer,
                 param_name='lr'  # optional
             )
+            # We need to close the logger when we are done
```
Comment (vfdev-5) [inline ignite/contrib/handlers/polyaxon_logger.py:90]: I think it is optional and not required by plx

**Comment by sdesrozis** [comment]
@ahmedo42 Thank you! LGTM

**Review by sdesrozis — APPROVED** [review]
@ahmedo42 Thank you! LGTM

**Pull request merged**